src/Command/ReferenceCommand.php

```diff
@@ -13,111 +13,95 @@
-            $this->configuration->get('reference.namespace'),
-            str_replace(['/', '.php'], ['\\', ''], preg_replace(sprintf('#^%s/#i', $this->configuration->get('reference.src')), '', $fileName))
-        );
-        $content = '';
+        $templateFileName = 'reference.*.twig';
+        $templateContext = ['class' => new ClassParser($reflectionClass)];
 
-        $this->reflectionClass = new \ReflectionClass($namespace);
-        $outputFile = $input->getArgument('output');
+        if ($reflectionClass->implementsInterface(ConfigurationInterface::class)) {
+            $yaml = (new YamlReferenceDumper())->dump($reflectionClass->newInstance());
+            if (!$yaml) {
+                $style->error(sprintf('No configuration available in "%s".', $file->getPathname()));
 
-        if ($this->reflectionClass->implementsInterface(ConfigurationInterface::class)) {
-            return $this->generateConfigExample($stderr, $outputFile);
+                return self::FAILURE;
+            }
+
+            $templateFileName = 'configuration.*.twig';
+            $templateContext['configuration'] = $yaml;
         }
 
-        $content = $this->outputFormatter->writePageTitle($this->reflectionClass, $content);
-        $content = $this->outputFormatter->writeClassName($this->reflectionClass, $content);
-        $content = $this->reflectionHelper->handleParent($this->reflectionClass, $content);
-        $content = $this->reflectionHelper->handleImplementations($this->reflectionClass, $content);
-        $content = $this->phpDocHelper->handleClassDoc($this->reflectionClass, $content);
-        $content = $this->reflectionHelper->handleClassConstants($this->reflectionClass, $content);
-        $content = $this->reflectionHelper->handleProperties($this->reflectionClass, $content);
-        $content = $this->reflectionHelper->handleMethods($this->reflectionClass, $content);
+        $content = $this->environment->render(
+            $this->getTemplateFile($input->getOption('template-path'), $templateFileName)->getFilename(),
+            $templateContext
+        );
 
-        if (!$outputFile) {
-            fwrite(\STDOUT, $content);
-            $stderr->success('Reference successfully printed on stdout for '.$fileName);
+        if (!$input->getArgument('output')) {
+            $style->block($content);
 
-            return Command::SUCCESS;
+            return self::SUCCESS;
         }
 
-        if (!fwrite(fopen($outputFile, 'w'), $content)) {
-            $stderr->error('Error opening or writing '.$outputFile);
-
-            return Command::FAILURE;
+        $dirName = pathinfo($input->getArgument('output'), \PATHINFO_DIRNAME);
+        if (!is_dir($dirName)) {
+            mkdir($dirName, 0777, true);
         }
-        $stderr->success('Reference successfully generated for '.$fileName);
-
-        return Command::SUCCESS;
-    }
+        if (!file_put_contents($input->getArgument('output'), $content)) {
```

Comment:
`$input->getArgument('output')` should probably be in a variable to avoid calling it multiple times